control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -659,6 +676,12 @@ func (r *HostedControlPlaneReconciler) update(ctx context.Context, hostedControl
 	if err := r.reconcileInfrastructure(ctx, hostedControlPlane, createOrUpdate); err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to ensure infrastructure: %w", err)
 	}
+
+	r.Log.Info("Reconciling default security group")
```

Comment:
By calling this here before r.reconcile, we gate our HC availability on the creation of this SG being successful.
Wouldn't it make sense to call it after invoking r.reconcile so it does not preclude the HC from going available? Same applies for machine approver and autoscaling actually (but let's discuss those separately).

Reply:
Well, it doesn't really try to create the SG until we have a valid OIDC token, which only happens once we have a running KAS. 
However, I don't see the need for any of these calls to block each other, so I think it makes sense to collect a set of errors and report those as an aggregate. Will update.